chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/component/ChaosMonkeyScheduler.java

```diff
@@ -1,49 +1,83 @@
 package de.codecentric.spring.boot.chaos.monkey.component;
 
+import de.codecentric.spring.boot.chaos.monkey.assaults.ChaosMonkeyRuntimeAssault;
 import de.codecentric.spring.boot.chaos.monkey.configuration.AssaultProperties;
+import java.util.*;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.lang.Nullable;
 import org.springframework.scheduling.config.CronTask;
 import org.springframework.scheduling.config.ScheduledTask;
 import org.springframework.scheduling.config.ScheduledTaskRegistrar;
 
+/**
+ * @author Maxime Bouchenoire
+ * @author Lukas Morawietz
+ */
 public class ChaosMonkeyScheduler {
 
   private static final Logger Logger = LoggerFactory.getLogger(ChaosMonkeyScheduler.class);
 
-  @Nullable private final ScheduledTaskRegistrar scheduler;
+  private final ScheduledTaskRegistrar scheduler;
 
   private final AssaultProperties config;
+  private final List<ChaosMonkeyRuntimeAssault> assaults;
 
-  private final ChaosMonkeyRuntimeScope runtimeScope;
-
-  @Nullable private ScheduledTask currentTask;
+  private final Map<ChaosMonkeyRuntimeAssault, ScheduledTask> currentTasks = new HashMap<>();
 
   public ChaosMonkeyScheduler(
       ScheduledTaskRegistrar scheduler,
       AssaultProperties config,
-      ChaosMonkeyRuntimeScope runtimeScope) {
+      List<ChaosMonkeyRuntimeAssault> assaults) {
     this.scheduler = scheduler;
     this.config = config;
-    this.runtimeScope = runtimeScope;
+    this.assaults = assaults;
 
     reloadConfig();
   }
 
   public void reloadConfig() {
-    String cronExpression = config.getRuntimeAssaultCronExpression();
-    boolean active = !"OFF".equals(cronExpression);
-
-    if (currentTask != null) {
-      Logger.info("Cancelling previous task");
-      currentTask.cancel();
-      currentTask = null;
+    Map<ChaosMonkeyRuntimeAssault, String> cronExpressions = new LinkedHashMap<>();
```

Comment:
Its hard to follow the code in the method. Splitting the code into multiple methods might help. 

1. load expressions
2. filter/removing expressions which are already running/scheduled
3. cancel tasks which needs to be rescheduled
4. reschedule tasks

Currently 2 and 3 is combined, it might be not the best thing (maybe it is, but I would like to compare it how it looks)